pkg/math/math.go

```diff
@@ -51,3 +52,33 @@ func BigFloatToBigInt(f *big.Float, decimals uint64) *big.Int {
 
 	return result
 }
+
+// Float64StringToBigFloat converts a float64 string to a big.Float.
+func Float64StringToBigFloat(s string, decimals uint64) (*big.Float, error) {
```

Comment:
Are we going to remove the `BigInt` functions?  

Or can we express them in terms of the `BigFloat` functions if we keep them

Reply:
I think we keep the BigInt functions till we have everything converted to BigFloat.

Reply:
Why does this take  decimals? Any time we're scaling a value by an exponent, we have to be aware of the scaling factor in subsequent arithmetic operators, IMO moving to `big.Float` lets us avoid this